### ISSUE TITLE
Toward Characteristic-Preserving Image-based Virtual Try-On Network

### ISSUE THREAD

**Issue by reyllama** [reyllama/paper-reviews#163]
## TL; DR

- Propose CNN-based learnable Geometric Matching Module (GMM) that better warps the target clothing with less loss of details.

## Motivation

- Shape context matching with TPS was inefficient in ViTON, losing details of clothing during the warping process.

## Prior Works

- ViTON: clothing agnostic person representation + two stage generation framework

### Clothing-agnostic person representation

- Same as ViTON

![image](https://user-images.githubusercontent.com/46308407/160339138-be4c6e68-890b-4ef6-a98f-89793900f158.png)

### Geometric Matching Module

- Two similar-structured networks extract features from p (person repr.) and c (target clothes)
- Correlation matching layer combines the features into a single tensor (*not specified in the main paper*)
- Regression network predicts parameters for Thin Plate Spline. (offsets of size 2x5x5=50)
- TSP transforms the given cloth image into warped cloth image that is roughly aligned with the target (in pose, shape, etc.)
- Trained on L1 loss w.r.t. GT. (*Not directly stated, but assumably extracted from off-the-shelf segmentation model*)

![image](https://user-images.githubusercontent.com/46308407/160339975-4c449d1f-880c-441d-b884-473ea5978f45.png)

### Try-on Module

- Largely identical to that proposed in original VITON paper.
- Takes person repr. and warped cloth (*this is different*) as input, and first outputs composition mask as well as coarse result.
- Blend warped cloth with coarse result using composition mask so as to preserve the details of the given cloth.

![image](https://user-images.githubusercontent.com/46308407/160340229-a2620d49-cd9e-4cd6-8d55-f2811efd5f79.png)

### Evaluation protocol

- Split clothes into those with large TV (complex patterns) and those with small TV (plain pattern)
- Aim to demonstrate *ours* outperforms in **Large** and is on par in **Small**.

![image](https://user-images.githubusercontent.com/46308407/160340260-58c869cd-1f0d-467e-b7dd-1933d8434224.png)

### Geometric Matching Module

![image](https://user-images.githubusercontent.com/46308407/160340455-d23801c4-5877-4755-8849-f9a5228bb8b8.png)

### Baseline Comparison

![image](https://user-images.githubusercontent.com/46308407/160340531-9b7d2922-0195-4618-a355-70dc42995fe6.png)
![image](https://user-images.githubusercontent.com/46308407/160340552-ff7f41dc-0dc1-41ff-8421-ed172a850844.png)

- Note each column compares *ours* to the baseline (higher is better, as in *preferred*)

### Ablation

![image](https://user-images.githubusercontent.com/46308407/160340656-a421ed6a-38c4-49a5-aa16-c818962d66d7.png)

### Analysis

![image](https://user-images.githubusercontent.com/46308407/160340697-7dcc11cc-81a3-4d60-9d15-a40216e3d25a.png)
![image](https://user-images.githubusercontent.com/46308407/160340712-f8c01dd5-f95f-4cfd-91c5-2d907c5a94db.png)

## Comments

- Simple yet quite effective modification to the seminal work of VITON.
